portal-kernel/src/com/liferay/portal/kernel/repository/DocumentRepository.java

```diff
@@ -149,16 +147,8 @@ public default FileEntry getFileEntryByExternalReferenceCode(
 			String externalReferenceCode)
 		throws PortalException {
 
-		try {
-			return getFileEntry(
-				GetterUtil.getLongStrict(externalReferenceCode));
-		}
-		catch (NumberFormatException numberFormatException) {
-			throw new NoSuchFileEntryException(
-				"No file entry exists with external reference code " +
-					externalReferenceCode,
-				numberFormatException);
-		}
+		throw new UnsupportedOperationException(
+			"This method needs to be implemented");
```

Comment:
Instead of having a default implementation that throws an exception, why not keep the method fully abstract (w/o deflt impl)?

Reply:
Regarding as we have discussed, we will throw two UnsopportedException in `getFileEntryByExternalReferenCode` and `fetchFileEntryByExternalReferenCode`